tests/system/aiplatform/test_e2e_tabular.py

```diff
@@ -0,0 +1,159 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import os
+import uuid
+from tests.system.aiplatform.e2e_base import TestEndToEnd
+from urllib import request
+
+from google.cloud import aiplatform
+
+
+_PROJECT = os.getenv("BUILD_SPECIFIC_GCLOUD_PROJECT")
+_LOCATION = "us-central1"
+
+_BLOB_PATH = "california-housing-data.csv"
+_DATASET_SRC = "https://dl.google.com/mlcc/mledu-datasets/california_housing_train.csv"
+_DIR_NAME = os.path.dirname(os.path.abspath(__file__))
+_LOCAL_TRAINING_SCRIPT_PATH = os.path.join(
+    _DIR_NAME, "test_resources/california_housing_training_script.py"
+)
+
+
+@pytest.mark.usefixtures("prepare_staging_bucket", "delete_staging_bucket", "teardown")
+class TestEndToEndTabular(TestEndToEnd):
+    """End to end system test of the Vertex SDK with tabular data adapted from
+    reference notebook http://shortn/_eyoNx3SN0X"""
+
+    _temp_prefix = "temp-vertex-sdk-e2e-tabular"
+
+    def test_end_to_end_tabular(self, shared_state):
+        """Build dataset, train a custom and AutoML model, deploy, and get predictions"""
+
+        assert shared_state["bucket"]
+        bucket = shared_state["bucket"]
+
+        blob = bucket.blob(_BLOB_PATH)
+
+        # Download the CSV file into memory and save it directory to staging bucket
+        with request.urlopen(_DATASET_SRC) as response:
+            data = response.read()
+            blob.upload_from_string(data)
+
+        # Collection of resources generated by this test, to be deleted during teardown
+        shared_state["resources"] = []
+
+        aiplatform.init(
+            project=_PROJECT,
+            location=_LOCATION,
+            staging_bucket=shared_state["staging_bucket_name"],
+        )
+
+        # Create and import to single managed dataset for both training jobs
+
+        ds = aiplatform.TabularDataset.create(
+            display_name=f"{self._temp_prefix}-dataset-{uuid.uuid4()}",
+            gcs_source=[f'gs://{shared_state["staging_bucket_name"]}/{_BLOB_PATH}'],
+            sync=False,
+        )
+
+        shared_state["resources"].extend([ds])
+
+        # Define both training jobs
+
+        custom_job = aiplatform.CustomTrainingJob(
+            display_name=f"{self._temp_prefix}-train-housing-custom-{uuid.uuid4()}",
+            script_path=_LOCAL_TRAINING_SCRIPT_PATH,
+            container_uri="gcr.io/cloud-aiplatform/training/tf-cpu.2-2:latest",
+            requirements=["gcsfs==0.7.1"],
+            model_serving_container_image_uri="gcr.io/cloud-aiplatform/prediction/tf2-cpu.2-2:latest",
+        )
+
+        automl_job = aiplatform.AutoMLTabularTrainingJob(
+            display_name=f"{self._temp_prefix}-train-housing-automl-{uuid.uuid4()}",
+            optimization_prediction_type="regression",
+            optimization_objective="minimize-rmse",
+            optimization_objective_recall_value=None,
```

Comment:
We can remove `optimization_objective_recall_value` and `optimization_objective_precision_value`